dockers/docker-sonic-telemetry/certificate_rotation_checker

```diff
@@ -0,0 +1,214 @@
+    directory which stores the SONiC credentials and reloads telemetry server
+    configuration if its certificate was rotated.
+
+
+    Args:
+        certificate_file_name: A string indicates the telemetry certificate file name.
+
+    Returns:
+        None.
+    """
+    certificate_file_rotated = False
+
+    inotify_instance = inotify.adapters.Inotify()
+    inotify_instance.add_watch(CREDENTIALS_DIR_PATH)
+    for event in inotify_instance.event_gen(yield_nones=False):
+        header, event_type, monitoring_path, file_name = event
+        if (file_name == certificate_file_name
+                and ("IN_CREATE" in event_type or "IN_MOVED_TO" in event_type)):
+            certificate_file_rotated = True
+
+        if certificate_file_rotated:
+            certificate_file_rotated = False
+            syslog.syslog(syslog.LOG_INFO,
+                          "Certificate was rotated and reloading telemetry server configuration ...")
+
+            if not reload_telemetry_server_configuration():
+                syslog.syslog(syslog.LOG_ERR,
+                              "Failed to reload the telemetry server configuration!")
+
+            syslog.syslog(syslog.LOG_INFO, "Telemetry server configuration was reloaded successfully!")
+
+
+def certificate_rotated_checker():
+    """Checks rotation of certificate file and then reloads streaming telemetry server configuration.
+
+    Leverages 'inotify' module to check whether the certificate file of streaming telemetry was
+    rotated or not. The configuration of telemetry server process will be reloaded if it was rotated.
+
+    Args:
+        None
+
+    Returns:
+        None
+    """
+    certificate_file_path = ""
+    private_key_file_path = ""
+    certificate_file_name = ""
+
+    config_db = swsscommon.DBConnector("CONFIG_DB", 0)
+    telemetry_table = swsscommon.Table(config_db, "TELEMETRY")
+    telemetry_table_keys = telemetry_table.getKeys()
+    if "certs" in telemetry_table_keys:
+        certs_info = dict(telemetry_table.get("certs")[1])
+        if "server_crt" in certs_info and "server_key" in certs_info:
+            certificate_file_path = certs_info["server_crt"]
+            private_key_file_path = certs_info["server_key"]
+            syslog.syslog(syslog.LOG_INFO, "Path of certificate file is '{}'".format(certificate_file_path))
+            syslog.syslog(syslog.LOG_INFO, "Path of private key file is '{}'".format(private_key_file_path))
+        else:
+            syslog.syslog(syslog.LOG_ERR,
+                          "Failed to retrieve the path of certificate or private key file from 'TELEMETRY' table!")
+            sys.exit(1)
+    else:
+        syslog.syslog(syslog.LOG_ERR,
+                      "Failed to retrieve the certificate information from 'TELEMETRY' table!")
+        sys.exit(2)
+
+    while True:
+        if not os.path.exists(certificate_file_path) or not os.path.exists(private_key_file_path):
+            syslog.syslog(syslog.LOG_ERR,
```

Comment:
>LOG_ERR [](http://example.com/codeflow?start=33&length=7)

Instead of error message and keep loop, it is better to just simply kill telemetry daemon.